state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -760,26 +655,11 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		}
 	}
 
-	pub fn take_awaiting_broadcast(
-		broadcast_attempt_id: BroadcastAttemptId,
-	) -> Option<BroadcastAttempt<T, I>> {
-		if let Some(signing_attempt) = AwaitingBroadcast::<T, I>::take(broadcast_attempt_id) {
-			assert_eq!(
-				signing_attempt.broadcast_attempt.broadcast_attempt_id,
-				broadcast_attempt_id,
-				"The broadcast attempt id of the signing attempt should match that of the broadcast attempt id of its key"
-			);
-			Some(signing_attempt.broadcast_attempt)
-		} else {
-			None
-		}
-	}
-
 	pub fn remove_pending_broadcast(broadcast_id: &BroadcastId) {
 		PendingBroadcasts::<T, I>::mutate(|pending_broadcasts| {
 			if !pending_broadcasts.remove(broadcast_id) {
-				cf_runtime_utilities::log_or_panic!(
-					"The broadcast_id should exist in the pending broadcasts list since we added it to the list when the broadcast was initated"
+				log::warn!(
+					"The broadcast_id already aborted, but is now reported as successful. broadcast_id: {}", broadcast_id
```

Comment:
nit: this error message makes some strong assumptions about the calling context...

How do we know that it's not the other way round: a previously successful broadcast is now reported as failed? Or a failed broadcast was reported as failed again?

Reply:
(changed this)